client/hwlib/src/collectors/os_info.rs

```diff
@@ -0,0 +1,126 @@
+/* Copyright 2024 Canonical Ltd.
+ * All rights reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Written by:
+ *        Nadzeya Hutsko <nadzeya.hutsko@canonical.com>
+ */
+
+use regex::Regex;
+use std::fs;
+use std::process::Command;
+
+use crate::constants::PROC_VERSION_FILEPATH;
+use crate::models::software;
+
+pub fn collect_os_info(
+    proc_version_filepath: Option<&'static str>,
+) -> Result<software::OS, Box<dyn std::error::Error>> {
+    let codename = super::os_info::get_codename()?;
```

Comment:
I'd suggest importing the functions from `os_info` rather than use `super` here.

Reply:
Fixed in fed34d